ui/lib/apps/TopSql/pages/TopSql.tsx

```diff
@@ -0,0 +1,225 @@
+    setTimeRange(recentTimeRange)
+  }, [])
+
+  return (
+    <div className={styles.container}>
+      <Card>
+        <Space size="middle">
+          <InstanceSelect value={instanceId} onChange={setInstanceId} />
+          <Button.Group>
+            <TimeRangeSelector
+              value={timeRange}
+              onChange={handleSetTimeRange}
+            />
+            <Button icon={<ZoomOutOutlined />} onClick={handleZoomOut} />
+          </Button.Group>
+          <AutoRefreshButton
+            autoRefreshSeconds={autoRefreshSeconds}
+            onAutoRefreshSecondsChange={handleAutoRefreshSecondsChange}
+            remainingRefreshSeconds={remainingRefreshSeconds}
+            isLoading={isLoading}
+            onRefresh={refreshTimestampRange}
+            options={autoRefreshOptions}
+          />
+        </Space>
+      </Card>
+      <Spin spinning={isLoading}>
+        {!isLoading && !seriesData?.length && (
+          <p style={{ marginTop: '100px', textAlign: 'center' }}>
+            {t('top_sql.no_data')}
+          </p>
+        )}
+        <div className={styles.chart_container}>
+          <TopSqlChart
+            onBrushEnd={handleBrushEnd}
+            seriesData={seriesData}
+            timeRange={timeRange}
+            timestampRange={queryTimestampRange}
+          />
+        </div>
+        {!!seriesData?.length && <TopSqlTable data={seriesData} />}
+      </Spin>
+    </div>
+  )
+}
+
+const queryTopSQLDigests = asyncDebounce(
+  (
+    instanceId: InstanceId,
+    windowSize: number,
+    timeRange: TimeRange,
+    topN: string
+  ) => {
+    const [beginTs, endTs] = calcTimeRange(timeRange)
+    return client
+      .getInstance()
+      .topsqlCpuTimeGet(
+        endTs as any,
+        instanceId,
+        beginTs as any,
+        topN,
+        `${windowSize}s` as any
+      )
+      .then((data) => {
+        const seriesData = data?.data.data
+        if (seriesData) {
+          // Sort data by digest
+          // If this digest occurs continuously on the timeline, we can easily see the sequential overhead
+          seriesData.sort(
+            (a, b) => a.sql_digest?.localeCompare(b.sql_digest!) || 0
+          )
+          seriesData.forEach((d) => {
+            convertOthersRecord(d)
+            d.plans?.forEach(
+              (item) =>
+                (item.timestamp_secs = item.timestamp_secs?.map(
+                  (t) => t * 1000
+                ))
+            )
+          })
+        }
+        return {
+          seriesData,
+          queryTimestampRange: [beginTs * 1000, endTs * 1000] as [
+            number,
+            number
+          ],
+        }
+      })
+  },
+  100,
+  { leading: false }
+)
+
+function useSeriesData(
+  instanceId: InstanceId,
+  timeRange: TimeRange,
+  autoRefreshSeconds: number,
+  refreshTimestamp: number
+) {
+  const { windowSize } = useWindowSize()
+
+  const { data, ...otherReturns } = useQuery(
+    [
+      'getSeriesData',
```

Comment:
I don't think it is a good idea to introduce a new mechanism here, for two reasons:

1. You have only changed partially, without touching the rest. Which means, this refactor attempt is unfinished and there are remaining works. You should either create an issue to track your remaining works, or you should open a PR to do it once for all.

2. You want to introduce a new mechanism. But whether this mechanism is suitable (e.g. can be fluently applied to the rest of the code) has not come to an agreement using some preflight designs. For example, as you can see, it is very complex to invoke a single call in your code base, L204~L216. I can hardly tell that this is a solution that will be accepted without hesitation. --- So, your new approach is very easy to be challenged and change a lot without reaching agreements.